tsz-compress/src/deltadelta_zigzag.rs

```diff
@@ -0,0 +1,589 @@
+use crate::prelude::{BitBuffer, BitBufferSlice};
+
+pub fn encode_deltadelta_i8(value: i8, out: &mut BitBuffer) {
+    out.push(false);
+    if value == 0 || value == -1 {
+        // 1 bit, header 00
+        out.push(false);
+        out.push(false);
+        if value == 0 {
+            out.push(false);
+        }
+        else {
+            out.push(true);
+        }
+        return;
+    } else if (-15..15).contains(&value){
+        // 5 bits, header 01
+        out.push(false);
+        out.push(true);
+        let encoded: u8 = ((value << 1) ^ (value >> 7)) as u8;
+        for i in 0..5{
+            out.push(encoded & (1 << i) != 0);
+        }
+    } else {
+        // 9 bits, header 10
+        out.push(true);
+        out.push(false);
+        let encoded: u8 = ((value << 1) ^ (value >> 7)) as u8;
+        for i in 0..8{
+            out.push(encoded & (1 << i) != 0);
+        }
+        out.push(false);
+    }
+}
+
+pub fn encode_deltadelta_i16(value: i16, out: &mut BitBuffer) {
+    out.push(false);
```

Comment:
There are two things that will probably immediately lead to better performance.

1. Extend the buffer with a single call instead of multiple with values that may be known at compile time: https://docs.rs/bitvec/latest/bitvec/vec/struct.BitVec.html#method.extend_from_bitslice
2. Use a match statement with pattern matching ranges of integers (and extend the bitslice in one call within the match arm)